### DISCUSSION TITLE
Initializers "Update"

### DISCUSSION THREAD

**Discussion by srburton** [dotnet/csharplang#1639] [Language Ideas]
**My idea is that an upgrade initializer would be possible to save code lines**

Initializers 
```
var people  = new People{
   Name = "Renato"
}
```

//Update object 
```
with people{
   Age = "M",
  Date = "2018/06/15"
}
```

//Output
```
Renato
M
2018/06/15
```

**Comment by franchyd** [comment]
That code doesn't seem at all readable or clear.

**Comment by Opiumtm** [comment]
Maybe this could be better?

```C#
with people {
  Age = "M",
  Date = "2018/06/15"
};
```

**Comment by Unknown6656** [comment]
Do you mean the C# 8.0's candidate #162?

**Comment by HaloFour** [comment]
`with` expressions are different from the `with` statement that is frequently requested.  The former creates a new instance with the modified properties whereas the latter modifies an instance in place.

**Comment by srburton** [comment]
@Opiumtm   

Very good solution liked

`with people {
  Age = "M",
  Date = "2018/06/15"
};`

``

**Comment by stuartstein777** [comment]
@HaloFour 

> with expressions are different from the with statement that is frequently requested. The former creates a new instance with the modified properties whereas the latter modifies an instance in place.

I'd love the former, if it means I can have an immutable type and then get a new instance back with just the specified member changed.

**Comment by PathogenDavid** [comment]
Reminds me of the with statement in [Pascal](https://www.freepascal.org/docs-html/ref/refsu62.html) and [JavaScript](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/with#Examples).

Something like this would be interesting, although I think I'd prefer it only be available for setting fields and properties (similar to the already existing [object initializers](https://github.com/dotnet/csharplang/blob/7f39331672cf8edbda8867de004138e0f711c877/spec/expressions.md#object-initializers).) (As opposed to the Pascal and JavaScript implementations where you're essentially changing `this` for a particular scope.)

That being said, the number of times I've wanted something like this have been pretty limited.
